devito/propagator.py

```diff
@@ -308,7 +308,11 @@ def generate_loops(self, loop_body):
         omp_master = [cgen.Pragma("omp master")] if self.compiler.openmp else []
         omp_single = [cgen.Pragma("omp single")] if self.compiler.openmp else []
         omp_parallel = [cgen.Pragma("omp parallel")] if self.compiler.openmp else []
-        omp_for = [cgen.Pragma("omp for schedule(static)")] if self.compiler.openmp else []
+        reduction_list = ["kernel", "loop_body"] if self.compiler.openmp else []
+        omp_for = [cgen.Pragma("omp for schedule(static)"\
+                   + (self.profiler.get_loop_reduction("+", reduction_list)\
+                   if self.profile else ""))]\
+                  if self.compiler.openmp else []
```

Comment:
This should be managed by the Profiler as well, so that we can do a different reduction on timing since that doesn't work with OpenMP and `+` is not the right reduction here I believe.